src/ObjectBuilder.cs

```diff
@@ -124,7 +126,7 @@ public virtual IModelType BuildServiceType(string serviceTypeName)
                 else
                 {
                     enumType.ModelAsString = true;
-                    enumType.SetName( string.Empty);
+                    enumType.SetName(string.Empty);

```

Comment:
What happens here if the underlying enum type is a complex object? I doubt if we allow that in AutoRest today (should we?), I have opened an issue in the linter to warn against that [here](https://github.com/Azure/azure-openapi-validator/issues/107) but ideally we should throw here if we see anything non primitive
@fearthecowboy  thoughts?